modules/zenith-public/lua/4-additive/quests/repeatableQuestDoubleRewards.lua

```diff
@@ -0,0 +1,524 @@
+    },
+
+    [xi.zone.SELBINA] =
+    {
+        ['Falgima'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.A_POTTERS_PREFERENCE },
+        ['Melyon'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.FISHERMANS_HEART },
+        ['Valgeir'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.ONLY_THE_BEST },
+        ['Dortain'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.ORLANDOS_ANTIQUES },
+    },
+
+    [xi.zone.MHAURA] =
+    {
+        ['Take'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.AN_EXPLORERS_FOOTSTEPS },
+        ['Rycharde'] = { xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.CARGO },
+    },
+
+    [xi.zone.KAZHAM] =
+    {
+        ['Ronan'] = { xi.questLog.OUTLANDS, xi.quest.id.outlands.THE_FIREBLOOM_TREE },
+        ['Lalapp'] = { xi.questLog.OUTLANDS, xi.quest.id.outlands.GREETINGS_TO_THE_GUARDIAN },
+    },
+
+    [xi.zone.NORG] =
+    {
+        ['Aeka'] = { xi.questLog.OUTLANDS, xi.quest.id.outlands.BLACK_MARKET },
+    },
+
+    -----------------------------------
+    -- Adoulin
+    -----------------------------------
+    [xi.zone.WESTERN_ADOULIN] =
+    {
+        ['Gogosomi'] = { xi.questLog.ADOULIN, xi.quest.id.adoulin.CAFETERIA },
+        ['Epaunice'] = { xi.questLog.ADOULIN, xi.quest.id.adoulin.FLOWER_POWER },
+        ['Tibelda'] = { xi.questLog.ADOULIN, xi.quest.id.adoulin.GRANDDADDY_DEAREST },
+    },
+
+    [xi.zone.EASTERN_ADOULIN] =
+    {
+        ['Mapolisa'] = { xi.questLog.ADOULIN, xi.quest.id.adoulin.IT_NEVER_GOES_OUT_OF_STYLE },
+    },
+}
+
+-----------------------------------
+-- Helper: Check if quest is in whitelist
+-----------------------------------
+local function isRepeatableQuest(areaId, questId)
+    local areaQuests = repeatableQuests[areaId]
+    if areaQuests == nil then
+        return false
+    end
+
+    return areaQuests[questId] == true
+end
+
+-----------------------------------
+-- Helper: Get log ID from area parameter
+-----------------------------------
+local function getLogId(area)
+    if type(area) == 'number' then
+        return area
+    elseif area and area['quest_log'] then
+        return area['quest_log']
+    end
+
+    return nil
+end
+
+-----------------------------------
+-- Helper: Get daily completion count for a quest
+-----------------------------------
+local function getDailyCount(player, areaId, questId)
+    local varFullName = string.format('DailyGilBonus_%d_%d', areaId, questId)
+    return player:getCharVar(varFullName)
+end
+
+-----------------------------------
+-- Helper: Increment and save daily count with midnight expiry
+-----------------------------------
+local function incrementDailyCount(player, areaId, questId)
+    local currentCount = getDailyCount(player, areaId, questId)
+    local newCount = currentCount + 1
+    local nextMidnight = getVanaMidnight(0)
+    local varFullName = string.format('DailyGilBonus_%d_%d', areaId, questId)
```

Comment:
```suggestion
    local varFullName = getDailyVarName(areaId, questId)
```

and define a `local function` above